packagepoa/transform.py

```diff
@@ -121,34 +127,41 @@ def add_file_to_zipfile(new_zipfile, name, new_name):
         return
     new_zipfile.write(name, new_name)
 
-def copy_pdf_to_output_dir(file_title_map, output_dir, doi, current_zipfile, poa_config):
-    """
-    we will attempt to generate a headless pdf and move this pdf
-    to the output directory.
 
-    if this headless creation fails, we will raise an error in the log file
+def extract_pdf_from_zip(name, current_zipfile, decap_name, poa_config):
+    # we extract the pdf from the zipfile
+    file_from_zip = current_zipfile.read(name)
+    decap_name_plus_path = poa_config.get('tmp_dir') + "/" + decap_name
+    # we save the pdf to a local file
+    with open(decap_name_plus_path, "wb") as temp_file:
+        temp_file.write(file_from_zip)
+    return decap_name_plus_path
 
-    the function that we call to decapitate the pdf is contained in decapitate_pdf.py.
-    It manages some error handline, and tries to determine whether the pdf
-    cover content has been cleanly removed.
-    """
 
+def pdf_new_name(pdf_name, title, doi, poa_config):
+    new_name = gen_new_name_for_file(pdf_name, title, doi, poa_config.get('filename_pattern'))
+    LOGGER.info("new_name: %s", new_name)
+    return new_name
+
+
+def pdf_decap_name(new_name):
+    return "decap_" + new_name
+
+
+def pdf_details(file_title_map):
+    "find the PDF file name in the zip file map"
+    pdf_name = None
+    title = None
     for name in file_title_map.keys():
```

Comment:
Can omit `.keys()` call as it used by default for `dict` iteration.

Reply:
Personally I like ``.keys()`` or using ``for key, value in ...`` over a ``dict``, to indicate the type, however the linter doesn't like it latter, and if the former is the default, I will yield to popular opinion!